packages/client/wallets/smart-wallet/src/blockchain/wallets/account/signer.ts

```diff
@@ -0,0 +1,38 @@
+import type { EOASignerData, PasskeyDisplay, PasskeySignerData, SignerData, SignerDisplay } from "../../../types/API";
+
+export interface SignerConfig {
```

Comment:
Wrapping the data in a class let's us keep data specific helpers close w/ actual data, and avoid big switch statements.

Added so I could use `display` in a more type safe and ergonomic way.